Lib/test/test_tools/test_reindent.py

```diff
@@ -23,6 +23,13 @@ def test_help(self):
         self.assertEqual(out, b'')
         self.assertGreater(err, b'')
 
+    def test_reindent_file_with_bad_encoding(self):
+        bad_coding_path = os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), '../bad_coding.py')
```

Comment:
`os.path.join(..., os.pardir, 'bad_coding.py')`

It is common also to use `support.findfile()`.